pkg/providers/instancetype/types.go

```diff
@@ -137,6 +141,39 @@ func computeRequirements(ctx context.Context, info *ec2.InstanceTypeInfo, offeri
 	return requirements
 }
 
+func getOs(info *ec2.InstanceTypeInfo, family amifamily.AMIFamily) string {
+
+	//We don't know the AMIFamily. Suppose user need a linux system.
+	if family == nil {
+		return string(v1.Linux)
+	}
+
+	if _, ok := family.(*amifamily.Windows); !ok {
```

Comment:
We should maybe do direct checks here-

- Check if the AMI family requested is Linux. If so-
  - Check architecture if needed
  - Check supported instance type
If any is true, return `Linux`.
- Check if the AMI family supported is Windows. If so-
  - Check architecture
  - Check supported instance type
If any is true, return `Windows`.

By default, Linux would be returned.

Reply:
Refine the code like.

```
if isWindowsAmiFamily && isSupportedWindows(instanceType) {
  return Windows
}

return Linux

func isSupportedWindows(instanceType) bool {
  //check explicated instance type
  //check architecture
  //check instance family
}

```